ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -229,6 +230,128 @@ export const ConfirmDelete = ({onCancel, onConfirm}) => {
   </Fragment>;
 };
 
+export const ConfirmDeleteUnattachedPD = ({onConfirm, onCancel}) => {
+  const [deleting, setDeleting] = useState(false);
+
+  return <Fragment>
+    <div style={{display: 'flex'}}>
+      <ClrIcon style={{color: colors.warning, gridColumn: 1, gridRow: 1}} className='is-solid'
+               shape='exclamation-triangle' size='20'/>
+      <h3 style={{...styles.baseHeader, ...styles.bold, gridColumn: 2, gridRow: 1}}>Delete environment options</h3>
+    </div>
+
+    <div style={styles.confirmWarning}>
+      <h3 style={{...styles.baseHeader, ...styles.bold, gridColumn: 1, gridRow: 1}}>
+        <label ><input type= 'radio' checked={deleting === true} onChange={() => setDeleting(true)}/>
+          Delete persistent disk</label></h3>
+      <p style={{...styles.confirmWarningText, gridColumn: 1, gridRow: 2}}>
+        Deletes your persistent disk, which will also delete all files on the disk.
+      </p>
+      <p style={{...styles.confirmWarningText, gridColumn: 1, gridRow: 3}}>
+        If you want to permanently save some files from the disk before deleting it,
+        you will need to create a new cloud environment to access it. </p>
+    </div>
+      <div>
+        <div>If you want to save some files permanently, such as input data, analysis outputs,
+          or installed packages, move them to the workspace bucket.</div>
+        <div>Note: Jupyter notebooks are autosaved to the workspace bucket, and deleting your disk will not delete your notebooks.</div>
+    </div>
+    <FlexRow style={{justifyContent: 'flex-end'}}>
+      <Button
+          type='secondaryLight'
+          aria-label={'Cancel'}
+          style={{marginRight: '.6rem'}}
+          onClick={() => onCancel()}>
+        Cancel
+      </Button>
+      <Button
+          aria-label={'Delete'}
+          disabled={!deleting}
+          onClick={async() => {
+            setDeleting(true);
+            try {
+              await onConfirm();
+            } catch (err) {
+              setDeleting(false);
+              throw err;
+            }
+          }}>
+        Delete
+      </Button>
+    </FlexRow>
+  </Fragment>;
+};
+
+export const ConfirmDeleteRuntimeWithPD = ({onCancel, onConfirm}) => {
+  const [deleting, setDeleting] = useState(false);
+  const [selectedKeepPD, setKeepPD] = useState(true);
```

Comment:
nit: I would just call the state variable `keepPD` for consistency. Usually hooks are `[x, setX]`